exercises/robot-simulator/example.rs

```diff
@@ -0,0 +1,102 @@
+#[derive(PartialEq, Debug, Clone)]
+pub enum Direction {
+    North,
+    East,
+    South,
+    West,
+}
+
+impl Direction {
+    pub fn previous_clockwise(&self) -> Self {
+        match self {
+            &Direction::North => Direction::West,
+            &Direction::East => Direction::North,
+            &Direction::South => Direction::East,
+            &Direction::West => Direction::South,
+        }
+    }
+
+    pub fn next_clockwise(&self) -> Self {
+        match self {
+            &Direction::North => Direction::East,
+            &Direction::East => Direction::South,
+            &Direction::South => Direction::West,
+            &Direction::West => Direction::North,
+        }
+    }
+}
+
+struct Position {
+    x: i32,
+    y: i32,
+}
+
+impl Position {
+    fn new(x: i32, y: i32) -> Self {
+        Position { x: x, y: y }
+    }
+
+    fn advance(self, direction: &Direction) -> Self {
+        match direction {
+            &Direction::North => Self::new(self.x, self.y + 1),
+            &Direction::South => Self::new(self.x, self.y - 1),
+            &Direction::East => Self::new(self.x + 1, self.y),
+            &Direction::West => Self::new(self.x - 1, self.y),
+        }
+    }
+}
+
+pub struct Robot {
+    position: Position,
+    direction: Direction,
+}
+
+impl Robot {
+    pub fn new(x: i32, y: i32, d: &Direction) -> Self {
+        Robot::build(Position::new(x, y), d.clone())
+    }
+
+    fn build(position: Position, direction: Direction) -> Self {
+        Robot {
+            position: position,
+            direction: direction,
+        }
+    }
+
+    pub fn turn_right(self) -> Self {
```

Comment:
well, to be fair, making them `Copy` was not the only solution - could have only made them `Clone`, not `Copy`, and then explicitly call `clone()` on them. But I suggested `Copy` simply because https://doc.rust-lang.org/std/marker/trait.Copy.html said "Generally speaking, if your type *can* implement `Copy`, it should"